### PR TITLE
Batch Generator Augmentations

### PR DESCRIPTION
**Pull request by martham93**
add support for two random augmentations: flipping horizontally and flipping vertically

### CODE REVIEW

File: pyveda/frameworks/batch_generator.py [hunk 1]
```diff
@@ -12,13 +14,17 @@ class BaseGenerator():
...
+                    flip_horizontal=False, flip_vertical=False):
         self.cache = cache
         self.batch_size = batch_size
         self.index = 0
         self.shuffle = shuffle
         self.rescale = rescale
         self.on_epoch_end()
+        self.flip_h = flip_horizontal
+        self.flip_v = flip_vertical
+        self._applied_augs(self.flip_h, self.flip_v)
```
Comment (rjpolackwich) [inline pyveda/frameworks/batch_generator.py]: how come this function is called here? looks like it doesn't change any state, just returns something that's intended to be used for something else, but the function return not handled

File: pyveda/vedaset/store/vedabase.py [hunk 2]
```diff
@@ -35,7 +35,7 @@ def images(self):
     def labels(self):
         return self._vset._label_array_factory(self._node.hit_table, self._node.labels,  self._vset)
 
-    def batch_generator(self, batch_size, shuffle=True, **kwargs):
+    def batch_generator(self, batch_size, shuffle=True, rescale=False, flip_horizontal=False, flip_vertical=False, **kwargs):
         """
         Generatates Batch of Images/Lables on a VedaBase partition.
         #Arguments
```
Comment (chelm) [inline pyveda/vedaset/store/vedabase.py]: Can we add the flips to args here for docs?

File: pyveda/frameworks/batch_generator.py (hunk 1 above)
Comment (chelm) [inline pyveda/frameworks/batch_generator.py]: I think this needs to be on an image by image basis. If this is setup on init then the random switches are applied only once. So every image will be aug'd the same. In the apply_transforms, it should figure out, randomly, which augs to apply to every image. 

Some will be flipped_vertifcally in a batch and others flipper_horizontally and others not touched.

File: pyveda/frameworks/batch_generator.py [hunk 3]
```diff
@@ -12,13 +14,17 @@ class BaseGenerator():
...
         self.list_ids = np.arange(0, len(self.cache))
```
Comment (rjpolackwich) [inline pyveda/frameworks/batch_generator.py]: `np.arange` takes a number and makes a range, so 0 is redundant: `np.arange(len(self.cache))`

File: pyveda/frameworks/batch_generator.py (hunk 1 above)
Comment (rjpolackwich) [reply pyveda/frameworks/batch_generator.py]: well calling this here doesn't do anything at all except return an array that never get's used or dealt with. @martham93 don't the transforms get applied sample by sample? it looks like a loop over it in line 158

File: pyveda/frameworks/batch_generator.py [hunk 4]
```diff
@@ -87,7 +149,7 @@ def _data_generation(self, list_ids_temp):
         for i, _id in enumerate(list_ids_temp):
             x_img = self.cache.images[int(_id)]
             y_img = self.cache.labels[int(_id)]
-
+            x_img, y_img = self.apply_transforms(x_img, y_img)
```
Comment (rjpolackwich) [inline pyveda/frameworks/batch_generator.py]: here's where transforms applied?

Comment (martham93) [reply pyveda/frameworks/batch_generator.py]: yes

File: pyveda/vedaset/store/vedabase.py (hunk 2 above)
Comment (martham93) [reply pyveda/vedaset/store/vedabase.py]: yes added for docs in VedaStream and VedaBase

File: pyveda/frameworks/batch_generator.py (hunk 3 above)
Comment (martham93) [reply pyveda/frameworks/batch_generator.py]: fixed

File: pyveda/frameworks/batch_generator.py (hunk 1 above)
Comment (martham93) [reply pyveda/frameworks/batch_generator.py]: yeah since `self._applied_augs` isn't really don't anything there I've removed it.

Comment (martham93) [reply pyveda/frameworks/batch_generator.py]: so  `transforms = self._applied_augs(self.flip_h, self.flip_v)` happens in line 62 of `apply_transforms` and `self.apply_transforms` is called for VedaBase 150 and for VedaStream in 179

**Pull request merged**